JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -257,23 +332,98 @@ private fun LoadingContent(
  * @param isShowingErrors (state) whether the screen is showing errors or not
  * @param favorites (state) all favorites
  * @param onRefresh (event) request to refresh data
- * @param navigateToArticle (event) request navigation to Article screen
+ * @param navigateToExpandedArticle (event) request navigation to Article screen
  * @param onToggleFavorite (event) request a single favorite be toggled
  * @param modifier modifier for root element
  */
 @Composable
+@Suppress("UNUSED_PARAMETER") // TODO: Add method to navigate to expanded article
+
 private fun HomeScreenErrorAndContent(
     posts: List<Post>,
+    selectedPostId: String?,
+    useListDetail: Boolean,
+    isListOpen: Boolean,
     isShowingErrors: Boolean,
     favorites: Set<String>,
     onRefresh: () -> Unit,
-    navigateToArticle: (String) -> Unit,
+    navigateToExpandedArticle: (String) -> Unit,
     onToggleFavorite: (String) -> Unit,
-    modifier: Modifier = Modifier,
-    scrollState: LazyListState
+    onSelectPost: (String) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
+    listLazyListState: LazyListState,
+    detailLazyListStates: Map<String, LazyListState>,
+    modifier: Modifier = Modifier
 ) {
     if (posts.isNotEmpty()) {
-        PostList(posts, navigateToArticle, favorites, onToggleFavorite, modifier, scrollState)
+        val detailPost by derivedStateOf {
+            // TODO: Restructure the posts data to remove the magic 3 as a default
+            posts.find { it.id == selectedPostId } ?: posts[3]
+        }
+
+        Row {
+            if (useListDetail || isListOpen) {
+                PostList(
+                    posts = posts,
+                    onArticleTapped = onSelectPost,
+                    favorites = favorites,
+                    onToggleFavorite = onToggleFavorite,
+                    contentPadding = rememberInsetsPaddingValues(
+                        insets = LocalWindowInsets.current.systemBars,
+                        applyTop = false,
+                        applyEnd = !useListDetail,
+                    ),
+                    modifier = modifier
+                        .then(
+                            if (useListDetail) {
+                                Modifier.width(334.dp)
+                            } else {
+                                Modifier.fillMaxSize()
+                            }
+                        )
+                        .pointerInput(Unit) {
+                            while (currentCoroutineContext().isActive) {
+                                awaitPointerEventScope {
```

Comment:
How often does this happen? Can we optimize this in any way?

Reply:
I'm not sure... it should be pretty lightweight I'd hope?
If we really needed to optimize it, we could only install this modifier on the side that isn't currently active.